### PR TITLE
[NO JIRA]: Upgrade eslint-config-skyscanner to latest to support incomming TS support

### PR DESCRIPTION
**Pull request by olliecurtis**
This PR spilts out from #2368 and completes the linting separately before we merge the TS framework as this linter upgrade is needed for the TS framework support.

There is a lot of file changes with the vast majority of the changes being `prettier` adding `()` around parameters for functions.

### CODE REVIEW

File: packages/bpk-component-calendar/src/BpkCalendarNav.js [hunk 1]
```diff
@@ -37,31 +37,33 @@ import STYLES from './BpkCalendarNav.module.scss';
...
-  // Safeguard for disabled buttons is due to React bug in Chrome: https://github.com/facebook/react/issues/8308
-  // PR: https://github.com/facebook/react/pull/8329 - unresolved as of 22/12/2016
-  if (isWithinRange(month, { start: min, end: max })) {
-    event.persist();
-    callback(event, { month, source });
-  }
-};
+const changeMonth =
+  ({ callback, max, min, month, source }) =>
+  (event) => {
```
Comment (zgeor) [inline packages/bpk-component-calendar/src/BpkCalendarNav.js:42]: Not to be addressed in this PR: Do we need to review this issue? GH suggests the issue has been fixed

Comment (olliecurtis) [reply packages/bpk-component-calendar/src/BpkCalendarNav.js:42]: Yeah possibly, I notice that it might just be a browser thing or still an issue here: https://github.com/facebook/react/issues/7711

File: packages/bpk-component-drawer/src/BpkDrawerContent.js [hunk 2]
```diff
@@ -50,21 +50,21 @@ type Props = {
...
-    children,
-    closeOnScrimClick, // unusued from withScrim scrim HOC
-    isIphone, // unusued from withScrim scrim HOC
-    isIpad, // unusued from withScrim scrim HOC
+    hideTitle,
+    id,
+    isDrawerShown,
+    isIpad,
+    isIphone,
+    onClose, // unusued from withScrim scrim HOC
```
Comment (zgeor) [inline packages/bpk-component-drawer/src/BpkDrawerContent.js:65]: I think the reshuffle messed up the comments here. Also there is a typo in unusued

File: packages/bpk-component-icon/gulpfile.js [hunk 3]
```diff
@@ -29,4 +29,5 @@ require('@babel/register')({
   rootMode: 'upward',
 });
 
+// eslint-disable-next-line import/extensions
```
Comment (zgeor) [inline packages/bpk-component-icon/gulpfile.js:32]: Why is this needed? Can the import not be fixed?

File: packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js [hunk 4]
```diff
@@ -88,10 +88,10 @@ const InfiniteList = withInfiniteScroll(List);
       elementsPerScroll={5}
       initiallyLoadedElements={1}
       loaderIntersectionTrigger="small"
-      onScroll={evt => {}} // eslint-disable-line no-unused-vars
-      onScrollFinished={evt => {}} // eslint-disable-line no-unused-vars
+      onScroll={(evt) => {}} // eslint-disable-line no-unused-vars
```
Comment (zgeor) [inline packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js:91]: Will prepending an underscore to the var satisfy the lint rule?

Comment (olliecurtis) [reply packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js:91]: Are you referring to the `evt` param? I guess we could address this is a separate PR as this ignore has not been added in this PR. 

On the flipside this file will be deleted when we move to TS as its a flow test 😄

File: packages/bpk-stylesheets/build.js [hunk 5]
```diff
@@ -34,6 +34,7 @@ require('@babel/register')({
 
 const webpack = require('webpack');
 
+// eslint-disable-next-line import/extensions
```
Comment (zgeor) [inline packages/bpk-stylesheets/build.js:37]: Here too, is this fixable?

File: packages/bpk-component-icon/gulpfile.js (hunk 3 above)
Comment (olliecurtis) [reply packages/bpk-component-icon/gulpfile.js:32]: Will give it a go!

File: packages/bpk-component-flare/src/svgs/pointer.svg [hunk 6]
```diff
@@ -1 +1 @@
-<svg viewBox="0 0 14832 55"><path d="M7501.307 8.517l-68.043 39.341c-10.632 6.185-23.795 6.185-34.528 0l-68.144-39.34c-8.91-5.173-18.988-8.215-29.316-8.518H0v55h14832V0H7530.671a63.604 63.604 0 0 0-29.364 8.517z"/></svg>
\ No newline at end of file
+<svg viewBox="0 0 14832 55"><path d="M7501.307 8.517l-68.043 39.341c-10.632 6.185-23.795 6.185-34.528 0l-68.144-39.34c-8.91-5.173-18.988-8.215-29.316-8.518H0v55h14832V0H7530.671a63.604 63.604 0 00-29.364 8.517z"/></svg>
```
[the remark this answers is not in the document]
Comment (olliecurtis) [reply packages/bpk-component-flare/src/svgs/pointer.svg:1]: Prettier is not, I think a dep of a dep has been bumped here and the generated output has been updated so not linked directly to eslint or prettier making changes but `npm build` output

File: packages/bpk-component-drawer/src/BpkDrawerContent.js (hunk 2 above)
Comment (olliecurtis) [reply packages/bpk-component-drawer/src/BpkDrawerContent.js:65]: Ah yes reshuffle doesn't move comments 🤦 Good spot with the mis-spell (not me but good catch from years ago)

File: packages/bpk-stylesheets/build.js (hunk 5 above)
Comment (olliecurtis) [reply packages/bpk-stylesheets/build.js:37]: Will give it a go!

File: packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js (hunk 4 above)
Comment (zgeor) [reply packages/bpk-component-infinite-scroll/src/withInfiniteScroll-test.flow.js:91]: ah true, ignore

File: packages/bpk-component-flare/src/svgs/pointer.svg (hunk 6 above)
[the remark this answers is not in the document]
Comment (olliecurtis) [reply packages/bpk-component-flare/src/svgs/pointer.svg:1]: I also did a check and the pointer still renders as expected so at least the generated build output still is ok

[the remark this answers is not in the document]
Comment (olliecurtis) [reply packages/bpk-component-flare/src/svgs/pointer.svg:1]: Have managed to find a dep of a dep caused this but now have the fix to put it back :D

File: .flowconfig [hunk 7]
```diff
@@ -2,6 +2,9 @@
 .*/node_modules/jsonlint/.*
 .*/node_modules/config-chain/test/broken.json
 
+# This is expected as this malformed file is for a test in the library.
+.*/node_modules/resolve/test/resolver/malformed_package_json/package.json
```
Comment (olliecurtis) [inline .flowconfig:6]: Had to add this ignore here to the flowconfig as flow was throuwing an error on an external dependency which is expected as the file is part of a test.

**Review by zgeor — APPROVED** [review]
